packages/deployments/contracts/contracts/core/messaging/connectors/BSCConnector.sol

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.15;
+
+import {IRootManager} from "../interfaces/IRootManager.sol";
+
+import {Connector} from "./Connector.sol";
+
+/**
+ */
+interface MultichainCall {
+  function anyCall(
+    address _to,
+    bytes calldata _data,
+    address _fallback,
+    uint256 _toChainID,
+    uint256 _flags
+  ) external;
+
+  function context()
+    external
+    view
+    returns (
+      address from,
+      uint256 fromChainID,
+      uint256 nonce
+    );
+
+  function executor() external view returns (address executor);
+}
+
+//AMB: MultichainCall(0xC10Ef9F491C9B59f936957026020C321651ac078) -  Chain agnostic
+abstract contract BaseBSCConnector is Connector {
+  address private immutable executor; // Is != amb, used only to retrieve sender context
+
+  // ============ Constructor ============
+  constructor(
+    uint32 _domain,
+    uint32 _mirrorDomain,
+    address _amb,
+    address _rootManager,
+    address _mirrorConnector,
+    uint256 _mirrorProcessGas,
+    uint256 _processGas,
+    uint256 _reserveGas
+  )
+    Connector(_domain, _mirrorDomain, _amb, _rootManager, _mirrorConnector, _mirrorProcessGas, _processGas, _reserveGas)
+  {
+    executor = MultichainCall(_amb).executor();
+  }
+
+  // ============ Public Fns ============
+
+  /**
+    @dev Receive function signature
+  */
+  function anyExecute(bytes memory _data) external returns (bool success, bytes memory result) {
+    _processMessage(msg.sender, _data); // msg.sender not used
+  }
+
+  function _verifySender(address _expected) internal override returns (bool) {
+    require(msg.sender == AMB, "!bridge");
+
+    (address from, uint256 fromChainId, ) = MultichainCall(executor).context();
+    return from == _expected; // TODO: convert chainId to domain and assert fromId==mirrorDomain
```

Comment:
it would be nice to build this mapping into a registry, but for now can we store it as an immutable property similar to `executor`?